parsers/julia.c

```diff
@@ -0,0 +1,1331 @@
+/*
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License version 2 or (at your option) any later version.
+*
+*   This module contains functions for generating tags for Julia files.
+*/
+
+/*
+*   INCLUDE FILES
+*/
+#include "general.h"	/* must always come first */
+
+#include <string.h>
+
+#include "keyword.h"
+#include "parse.h"
+#include "entry.h"
+#include "options.h"
+#include "read.h"
+#include "routines.h"
+#include "vstring.h"
+#include "xtag.h"
+
+/*
+*   MACROS
+*/
+#define MAX_STRING_LENGTH 256
+
+/*
+*   DATA DEFINITIONS
+*/
+typedef enum {
+    K_CONSTANT,
+    K_FUNCTION,
+    K_FIELD,
+    K_MACRO,
+    K_MODULE,
+    K_STRUCT,
+    K_TYPE,
+    K_IMPORT,
+    K_NONE
+} JuliaKind;
+
+static kindDefinition JuliaKinds [] = {
+    { true, 'c', "constant", "Constants"    },
+    { true, 'f', "function", "Functions"    },
+    { true, 'g', "field",    "Fields"       },
+    { true, 'm', "macro",    "Macros"       },
+    { true, 'n', "module",   "Modules"      },
+    { true, 's', "struct",   "Structures"   },
+    { true, 't', "type",     "Types"        },
+    { true, 'x', "unknown",  "Imported name"}
+};
+
+typedef enum {
+	TOKEN_NONE=0,         /* none */
+	TOKEN_WHITESPACE,
+	TOKEN_PAREN_BLOCK,
+	TOKEN_BRACKET_BLOCK,
+	TOKEN_CURLY_BLOCK,
+    TOKEN_OPEN_BLOCK,
+    TOKEN_CLOSE_BLOCK,
+    TOKEN_TYPE_ANNOTATION,
```

Comment:
As far as reading the code on GitHub web UI, I see misalignment of lines.

Reply:
If you use Emacs, what you have to do is simple: \C-x h \M-\. But I guess you may use Geany. So I don't know the quick way to adjust code alignments.  If you want, I will do it on behalf of you.